src/sentry_core.c

```diff
@@ -1169,3 +1169,69 @@ sentry_clear_crashed_last_run(void)
     sentry__options_unlock();
     return success ? 0 : 1;
 }
+
+void
+sentry_capture_minidump(const char *dump_path, sentry_value_t event)
+{
+    sentry_path_t *sentry_dump_path
+        = sentry__path_from_str_n(dump_path, strlen(dump_path));
+
+
+    SENTRY_WITH_OPTIONS (options) {
+        sentry__write_crash_marker(options);
+
+        sentry_envelope_t *envelope = NULL;
+        sentry__ensure_event_id(event, NULL);
+        envelope = sentry__envelope_new();
+        if (!envelope || !sentry__envelope_add_event(envelope, event)) {
+            sentry_envelope_free(envelope);
+            sentry_value_decref(event);
+        }
+
+        SENTRY_TRACE("adding attachments to envelope");
+        for (sentry_attachment_t *attachment = options->attachments; attachment;
+             attachment = attachment->next) {
+            sentry_envelope_item_t *item = sentry__envelope_add_from_path(
+                envelope, attachment->path, "attachment");
+            if (!item) {
+                continue;
+            }
+            sentry__envelope_item_set_header(item, "filename",
+#ifdef SENTRY_PLATFORM_WINDOWS
+                sentry__value_new_string_from_wstr(
+#else
+                sentry_value_new_string(
+#endif
+                    sentry__path_filename(attachment->path)));
+        }
+
+        sentry_envelope_item_t *item = sentry__envelope_add_from_path(
+            envelope, sentry_dump_path, "attachment");
+        if (item) {
+            sentry__envelope_item_set_header(item, "attachment_type",
+                sentry_value_new_string("event.minidump"));
+
+            sentry__envelope_item_set_header(item, "filename",
+#ifdef SENTRY_PLATFORM_WINDOWS
+                sentry__value_new_string_from_wstr(
+#else
+                sentry_value_new_string(
+#endif
+                    sentry__path_filename(sentry_dump_path)));
+        }
+
+        sentry__capture_envelope(options->transport, envelope);
+
+        // now that the envelope was written, we can remove the temporary
+        // minidump file
+        sentry__path_remove(sentry_dump_path);
```

Comment:
While this makes sense in the crash handler (including the comment), removing the dump file is probably not expected in the general case. I would provide a parameter `remove_dump_on_send` (or something similar) to make this deletion conditional on the client's request.